domain/storage/state/storagepool.go

```diff
@@ -470,11 +470,28 @@ func getStoragePoolUUID(
 	db domain.TxnRunner,
 	name string,
 ) (domainstorage.StoragePoolUUID, error) {
-	// TODO: !!!
-	return "", nil
+	inputArg := storagePool{Name: name}
```

Comment:
Does it make sense to just introduce new types specific to this call and not try and re-use others that are mixing concerns?